File: java/lib/src/main/java/com/svix/api/Ingest.java
// this file is @generated
package com.svix.api;

import com.svix.SvixHttpClient;

import lombok.Getter;

public class Ingest {
    private final SvixHttpClient client;

    @Getter private final IngestAuthentication authentication;
    @Getter private final IngestEndpoint endpoint;
    @Getter private final IngestSource source;

    public Ingest(SvixHttpClient client) {
        this.client = client;
        this.authentication = new IngestAuthentication(client);
        this.endpoint = new IngestEndpoint(client);
        this.source = new IngestSource(client);
    }
}

File: kotlin/lib/src/main/kotlin/Ingest.kt
// this file is @generated
package com.svix.kotlin

class Ingest(private val client: SvixHttpClient) {
    val authentication: IngestAuthentication = IngestAuthentication(client)

    val endpoint: IngestEndpoint = IngestEndpoint(client)

    val source: IngestSource = IngestSource(client)
}
